src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs

```diff
@@ -53,6 +54,17 @@ public HubConnectionContext(ConnectionContext connectionContext, TimeSpan keepAl
             _clientTimeoutInterval = clientTimeoutInterval.Ticks;
         }
 
+        internal StreamTracker StreamTracker
+        {
+            get
+            {
+                if (_streamTracker == null)
```

Comment:
Why is this lazy? Performance?

Reply:
Yes, Brennan wants to keep the number of server-side allocations per connection down. Most hubs won't use the new streaming feature, and so don't allocate a StreamTracker on them. Added a comment.